components/gantt/state.md

```diff
@@ -0,0 +1,1158 @@
+
+        if (foundItem != null)
+        {
+            foundItem.Title = item.Title;
+            foundItem.Start = item.Start;
+            foundItem.End = item.End;
+            foundItem.PercentComplete = item.PercentComplete;
+        }
+    }
+
+    private void DeleteItem(GanttDeleteEventArgs args)
+    {
+        var item = Data.FirstOrDefault(i => i.Id.Equals((args.Item as GanttTask).Id));
+
+        RemoveChildRecursive(item);
+    }
+
+    private void RemoveChildRecursive(GanttTask item)
+    {
+        var children = Data.Where(i => item.Id.Equals(i.ParentId)).ToList();
+
+        foreach (var child in children)
+        {
+            RemoveChildRecursive(child);
+        }
+
+        Data.Remove(item);
+    }
+}
+````
+````Service
+using Microsoft.JSInterop;
+using System.Text.Json;
+public class LocalStorage
+{
+    protected IJSRuntime JSRuntimeInstance { get; set; }
+
+    public LocalStorage(IJSRuntime jsRuntime)
+    {
+        JSRuntimeInstance = jsRuntime;
+    }
+
+    public ValueTask SetItem(string key, object data)
+    {
+        return JSRuntimeInstance.InvokeVoidAsync(
+            "localStorage.setItem",
+            new object[] {
+                key,
+                JsonSerializer.Serialize(data)
+            });
+    }
+
+    public async Task<T> GetItem<T>(string key)
+    {
+        var data = await JSRuntimeInstance.InvokeAsync<string>("localStorage.getItem", key);
+        if (!string.IsNullOrEmpty(data))
+        {
+            return JsonSerializer.Deserialize<T>(data);
+        }
+
+        return default;
+    }
+
+    public ValueTask RemoveItem(string key)
+    {
+        return JSRuntimeInstance.InvokeVoidAsync("localStorage.removeItem", key);
+    }
+}
+````
+
+<!-- ### Save and Load Gantt State in a WebAssembly application
+The [knowledge base article for saving the Gantt state in a WASM application] explains two ways of storing the `Gantt` state - through a custom controller and a custom service that calls the browser's LocalStorage. -->
+
+### Set Gantt Options Through State
+
+The Gantt state allows you to control the behavior of the Gantt programmatically - you can, for example, set sorts, filteres, expand hierarhical items.
+
+>tip The individual tabs below show how you can use the state to programmatically set the Gantt filtering, sorting and other features.
+
+>tip If you want to set an initial state to the Gantt, use a similar snippet, but in the `OnStateInit` event.
+
+<div class="skip-repl"></div>
+````Sorting
+@[template](/_contentTemplates/gantt/state.md#set-sort-from-code)
+````
+````FilterRow
+@[template](/_contentTemplates/gantt/state.md#filter-row-from-code)
+````
+````FilterMenu
+@[template](/_contentTemplates/gantt/state.md#filter-menu-from-code)
+````
+````Hierarchy
+@[template](/_contentTemplates/gantt/state.md#expand-hierarchy-from-code)
+````
+
+@[template](/_contentTemplates/gantt/state.md#filter-menu-default-filters)
+
+
+### Set Default (Initial) State
```

Comment:
This example is one of the fundamental ones and should be first. Loading and Saving the state are based on this one, so that example should come after this one.